components/curators-list.tsx

```diff
@@ -1,127 +1,67 @@
-import { changeLoadState } from "contexts/reducers/change-load";
+import { CuratorsListPaginated } from "types/api";
 
-import { Curator } from "interfaces/curators";
-
-import useApi from "x-hooks/use-api";
 import usePage from "x-hooks/use-page";
 
-interface CuratorsPages {
-  curators: Curator[];
-  page: number;
+interface CuratorsListProps {
+  curators: CuratorsListPaginated;
 }
 
-export default function CuratorsList({ inView }: { inView?: boolean }) {
-  const { query } = useRouter();
+export default function CuratorsList({ curators }: CuratorsListProps) {
   const { t } = useTranslation(["common", "council"]);
 
-  const [hasMore, setHasMore] = useState(false);
-  const [truncatedData, setTruncatedData] = useState(false);
-  const [isEmptyPage, setIsEmptyPage] = useState<boolean>();
-  const [curatorsPage, setCuratorsPage] = useState<CuratorsPages[]>([]);
-
-  const { searchCurators } = useApi();
-  const { state, dispatch } = useAppState();
-  const { page, nextPage, goToFirstPage } = usePage();
-
-  function handlerSearch() {
-    if (!state.Service?.network?.active || inView === false || !query?.chain) return;
-
-    dispatch(changeLoadState(true));
-
-    searchCurators({
-      isCurrentlyCurator: true,
-      networkName: state.Service?.network?.active?.name,
-      sortBy: "acceptedProposals",
-      order: "asc",
-      page,
-      chainShortName: query.chain.toString()
-    })
-      .then(({ rows, pages, currentPage }) => {
-        if (currentPage > 1) {
-          if (curatorsPage.find((el) => el.page === currentPage)) return;
-
-          const tmp = [...curatorsPage, { page: currentPage, curators: rows }];
+  const [curatorsList, setCuratorsList] = useState<CuratorsListPaginated>();
 
-          tmp.sort((pageA, pageB) => {
-            if (pageA.page < pageB.page) return -1;
-            if (pageA.page > pageB.page) return 1;
+  const { nextPage } = usePage();
 
-            return 0;
-          });
-          setCuratorsPage(tmp);
-        } else {
-          setCuratorsPage([{ page: currentPage, curators: rows }]);
-        }
-
-        setIsEmptyPage(rows.length > 0 ? false : true)
-        setHasMore(currentPage < pages);
-      })
-      .catch((error) => {
-        console.error("Error fetching issues", error);
-      })
-      .finally(() => {
-        dispatch(changeLoadState(false));
-      });
-  }
-
-  useEffect(handlerSearch, [page, state.Service?.network?.active, inView, query?.chain]);
+  const isListEmpty = !curatorsList?.count;
+  const hasMore = !isListEmpty && curatorsList?.currentPage < curatorsList?.pages;
 
   useEffect(() => {
-    if (page) {
-      const pagesToValidate = [...Array(+page).keys()].map((i) => i + 1);
-
-      setTruncatedData(!pagesToValidate.every((pageV) =>
-          curatorsPage.find((el) => el.page === pageV)));
-    }
-  }, [page, curatorsPage]);
-
-  if(inView !== null && inView === false) return null;
+    if (!curators) return;
```

Comment:
wouldn't it make sense to have this list come into the component _already parsed_?